### ISSUE TITLE
Error when clicking on the drop-down button of the DateEntry

### ISSUE THREAD

**Issue by j4321** [j4321/tkcalendar#44]
``` python
import tkinter as tk
import tkinter.ttk as ttk
import tkcalendar as tkc

def main():
    root = tk.Tk()

    test_frame = ttk.Frame(root)
    test_frame.pack(side='top', fill='both', expand=True)

    test_date_entry = tkc.DateEntry(test_frame)
    test_date_entry.pack()

    root.mainloop()

if __name__ == '__main__':
    main()
```
I try your example code above and when I click the drop-down-arrow nothing happens except I get the error message below. 
The text filed of DateEntry shows the date in ISO-format 2019-07-19 

Python 3.7.4, tkinter 8.6, Windows 10.

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "C:\Program Files\Python37\lib\tkinter\__init__.py", line 1705, in __call__
    return self.func(*args)
  File "C:\Users\Magnus\AppData\Roaming\Python\Python37\site-packages\tkcalendar\dateentry.py", line 207, in _on_b1_press
    self.drop_down()
  File "C:\Users\Magnus\AppData\Roaming\Python\Python37\site-packages\tkcalendar\dateentry.py", line 278, in drop_down
    date = self.parse_date(self.get())
  File "C:\Users\Magnus\AppData\Roaming\Python\Python37\site-packages\tkcalendar\calendar_.py", line 927, in parse_date
    return parse_date(date, self._properties['locale'])
  File "C:\Users\Magnus\AppData\Roaming\Python\Python37\site-packages\babel\dates.py", line 1168, in parse_date
    return date(year, month, day)
ValueError: day is out of range for month
```

_Originally posted by @magnuskson in https://github.com/j4321/tkcalendar/issues/41#issuecomment-513181904_

**Comment by j4321** [comment]
@magnuskson It seems that there is an issue with the locale. Can you tell me if setting explicitly the locale, like in the code below, fix things?

``` python
import tkinter as tk
from tkcalendar import DateEntry

root = tk.Tk()

tk.Label(root, text='Choose date').pack()

cal = DateEntry(root, locale='en_US')
cal.pack()
```
Could you also tell me what `locale.getdefaultlocale()` returns and what `cal.cget('locale')` returns when you don't set explicitly the locale when creating the DateEntry?

**Comment by magnuskson** [comment]
Yes, it's probably a locale issue.
It works with en_US (and fr_FR also works). But it doesn't work with my default sv_SE (YYYY-MM-DD for Sweden).

**Comment by j4321** [comment]
I have been able to reproduce the issue, it is not directly coming from tkcalendar but from the library babel which is used to convert date strings into datetime objects, but for some reasons fails for the Swedish locale. I have reported the bug upstream https://github.com/python-babel/babel/issues/657. 

tkcalendar needs babel to get the months and days in the right language and I would rather avoid to introduce a new dependency so I will wait a bit to see if babel's developers fix the bug. If they don't I will use the module dateparser instead.

**Comment by magnuskson** [comment]
Thanks for sorting this out.

(I find it strange that the ISO-format YYYY-MM-DD does not work in Babel. On the other hand Sweden is, as far as I know, the only country where this standard actually is used in reality.)

**Comment by j4321** [comment]
I have looked into babel's code and identified the origin of the issue: `babel.dates.parse_date()` is using the medium length date format to parse the date. In Swedish, the year is at the end in this format (e.g. 26 juli 2019) so babel understand YYYY in the short format as the day which obviously raises an error.

I have found a way around this and submitted a pull request to the babel project. I am not sure of the time it will take to get my proposal reviewed and hopefully accepted, so in the meantime I have  integrated the fix directly in `tkcalendar.calendar_.py`
